src/com/palmergames/bukkit/towny/event/executors/TownyActionEventExecutor.java

```diff
@@ -66,6 +77,50 @@ private static boolean isAllowedAction(Player player, Location loc, Material mat
 
 		return !event.isCancelled();
 	}
+	
+	/**
+	 * Towny's primary internal test to determine if something can explode
+	 * at the given location, based on Towny's plot permissions.
+	 * 
+	 * @param loc - Location being tested.
+	 * @return true if the explosion is allowed.
+	 */
+	private static boolean isAllowedExplosion(Location loc) {
+		boolean canExplode = false;
+		TownyWorld world = TownyAPI.getInstance().getTownyWorld(loc.getWorld().getName());
+		if (world == null)
+			canExplode = false;
+		else {
+		
+			if (TownyAPI.getInstance().isWilderness(loc)) {
+				/*
+				 * Handle occasions in the wilderness first.
+				 */
+				if (world.isForceExpl() || world.isExpl())
+					canExplode = true;
+				if (!world.isExpl())
+					canExplode = false;			
+			} else {
+				/*
+				 * Must be inside of a town.
+				 */
+				canExplode = TownyAPI.getInstance().getTownBlock(loc).getPermissions().explosion;			

```

Comment:
If town explosions are enabled, do all townblocks in the town force toggle the explosion perm for the townblock?

Reply:
Town explosions enabling should enable them on all non-changed townblocks. So it will be a per-plot setting.